### ISSUE TITLE
Despite deadline, class catalog cannot be closed to students (?)

### ISSUE THREAD

**Issue by betaveros** [learning-unlimited/ESP-Website#2402]
I'm not entirely sure if I'm missing something since it feels like we've done this recently, but we haven't been able to close the catalog to students. Closing `/Student/Catalog` does nothing. Looking at the code, there is a `catalog_student` with a permission decorator attached, but it is not an `aux_call`. It appears that it was meant to be called by the `catalog` method if the current user is a student, but this commit removes that logic:
https://github.com/learning-unlimited/ESP-Website/commit/9bfd5c2e3980e3b205f9d0e53cc487d026c16de2#diff-cd310f15494269c1d6790b375cf2001bR681

**Comment by benjaminjkraft** [comment]
Yeah, this was done because checking a deadline requires checking the current user, which prevents caching the whole page, and at least in the current setup it's important to be able to cache the whole page.  We might be able to do something where we explicitly only support non-user-specific deadlines for the catalog (perhaps with a separate `/admin_catalog`) which has user-specific deadlines but is not cached, but in any case that's additional code.

**Comment by Dhirenderchoudhary** [comment]
@willgearty can i work on it ?
